[instancer] Recalc avgcharwidth after instancing

qxliu76 opened the pull request:
Port from https://github.com/fonttools/fonttools/commit/3573cd906bafa46069b64c68e97ea69e62d1c03b

behdad commented on src/hb-ot-os2-table.hh:
```diff
@@ -209,6 +209,26 @@ struct OS2
+    }
+
+    unsigned avg_width = 0;
+    if (num) avg_width = (unsigned) roundf (total_width / num);
```
This looks a bit weird. You might as well write:
```
unsigned avg_width = num ? ... : 0;
```

behdad commented:
@qxliu76 Any idea why instancer-solver is failing in the distcheck bot? Does it require any fonts that are not added to `Makefile.am`?

qxliu76 commented:
> @qxliu76 Any idea why instancer-solver is failing in the distcheck bot? Does it require any fonts that are not added to `Makefile.am`?

uh...I think I need to add approx() on some expected output float values. I'll fix it.
Weird it's not caught in meson test, I just found some tests are not enabled as well: e.g test-priority-queue, test-multimap etc. Any idea why?

behdad commented:
> > @qxliu76 Any idea why instancer-solver is failing in the distcheck bot? Does it require any fonts that are not added to `Makefile.am`?
> 
> uh...I think I need to add approx() on some expected output float values. I'll fix it. Weird it's not caught in meson test,

Yeah looks like most tests are not included in meson `compiled_tests`?!


> I just found some tests are not enabled as well: e.g test-priority-queue, test-multimap etc. Any idea why?

No idea. Can you add them please?

qxliu76 commented:
> > > @qxliu76 Any idea why instancer-solver is failing in the distcheck bot? Does it require any fonts that are not added to `Makefile.am`?
> > 
> > 
> > uh...I think I need to add approx() on some expected output float values. I'll fix it. Weird it's not caught in meson test,
> 
> Yeah looks like most tests are not included in meson `compiled_tests`?!
> 
> > I just found some tests are not enabled as well: e.g test-priority-queue, test-multimap etc. Any idea why?
> 
> No idea. Can you add them please?

Fixed solver test. And following Makefile.sources, I moved these tests from noinst_programs to compiled_tests.
